learn/components/styling.md

```diff
@@ -3,4 +3,191 @@ title: Styling
 order: 5
 ---
 
-{% stub %}
+## (Introduction)
+
+- Web Components have powerful styling capabilities.
+  - _Let’s start off positive, even if it’s going to get a little
+    hairy._
+- Shadow DOM encapsulation makes Web Component styles generally easier
+  to author.
+  - Elements within a shadow tree cannot be directly selected by styles
+    declared outside of it. Conversely, elements outside of a shadow
+    tree cannot be directly selected by styles declared within it.
+  - Web Component authors need not worry about naming conflicts for CSS
+    classes outside of a shadow tree.
+  - Web Component authors can use less specific selectors without much
+    worry as there are generally less elements used within a shadow
+    tree.
+- Encapsulation is not isolation. Styles set outside of a shadow tree
+  can indirectly influence styles within a shadow tree.
+  - _Use this to transition to the next section..._
+
+## Shadow Hosts and CSS Inheritance
+
+It is very important to understand the role of a **shadow host**.
+
+### What is a Shadow Host
+
+- A **shadow host** is an element with a **shadow tree** and it is not
+  the **shadow root**. If your Web Component uses the Shadow DOM, the
+  element itself is a shadow host.
+  - The **shadow root** is a non-element node that cannot be selected
+    or styled since it’s featureless.
+- A **shadow host** is not a part of a **shadow tree**. It exists
+  _outside_ of a shadow tree and therefore cannot be selected directly
+  by styles declared within its shadow tree (i.e. referencing the tag
+  name with a type selector will not select the host — only nested
+  elements of that type). We can select it with the `:host`
+  pseudo-class and its functional counterpart `:host()` — more on that
+  later.
+  - `:host` when combined with relative selectors will be based on the
+    shadow root within the shadow tree and not in the light tree.
+- Being a part of the DOM outside of a shadow tree, the host element
+  can be subject to styles declared in that context.
+
+### CSS Inheritance
+
+- **A shadow host’s styles become the point of inheritance for all
```

Comment:
I wonder if it’s a safe summation to say that any property that inherits will inherit from the light Dom, and everything else is encapsulated. Are there any other ways that styles penetrate without explicit opt in via eg parts?

Reply:
> I wonder if it’s a safe summation to say that any property that inherits will inherit from the light Dom, and everything else is encapsulated.

I believe that would be correct. In this statement, I think I was trying to say something to the effect that even though the shadow host isn’t a part of the shadow tree (and the shadow root is not an element), the inherited styles still come from the shadow host. So, then whatever properties a top-level element have that inherit will inherit their value from the shadow host (also properties set to `inherit` in top-level elements in the shadow tree). I can figure out how to say all that simpler.